### ISSUE TITLE
deploy ice using chef cookbook

### ISSUE THREAD

**Issue by sojharo** [Cloudkibo/ice#10]

**Comment by sojharo** [comment]
http://gettingstartedwithchef.com/

First two chapters would be useful in this to understand chef. It is something new and looks interesting. I would try to finish these chapters tomorrow.

https://www.digitalocean.com/community/tutorials/how-to-create-simple-chef-cookbooks-to-manage-infrastructure-on-ubuntu

https://learn.chef.io/learn-the-basics/ubuntu/

Other two might be helpful as reference.

**Comment by jekram** [comment]
Asak
I am quite familiar with Chef and its cookbooks. Go ahead and learn it. However, there are alternate technology that is much better. I will guide you. Chef is an older technology that is not worth spending lot of time.
Allah Hafiz
Jawaid

Date: Sun, 31 May 2015 13:27:35 -0700

**Comment by sojharo** [comment]
Salaam,

I did some tutorials on ice and installed apache for php application using a cookbook as part of the tutorial. I have got some understanding.

This is a tool to automatize the server configurations. I am not looking further into this now and would try running ice server using this after today's meeting. Now I am going to look into CloudKibo switch video task.

**Comment by sojharo** [comment]
Also please let me know about the alternate you were talking about. I would try the chef for this now. In case, if I fail with this. I would look into alternates too.

**Comment by jekram** [comment]
Sojharo asak

The alternate to Chef is Docker.Docker is much more then Chef.

What Chef is doing here is creating an image file based on your code, the libraries and the configuration file.

Chef is a great tool that automated all these steps so instead of manually building the image. You have Chef and its recipes that you can use build them. In the last 8 years tools like Chef, Puppet have done a grew job. It is based on Ruby and has a very vibrant community.

However, in the last 2 years there is a newer technology"Docker". Which is much broader in scope and it also dose thinks like building images. 

There are number of vendors implementing"Docker" technology including Docker

**Comment by jekram** [comment]
Sojharo asak

The alternate to Chef is Docker.Docker is much more then Chef.

What Chef is doing here is creating an image file based on your code, the libraries and the configuration file.

Chef is a great tool that automated all these steps so instead of manually building the image. You have Chef and its recipes that you can use build them. In the last 8 years tools like Chef, Puppet have done a grew job. It is based on Ruby and has a very vibrant community.

However, in the last 2 years there is a newer technology"Docker". Which is much broader in scope and it also dose thinks like building images. 

There are number of vendors implementing"Docker" technology including Docker, Google, Red hat, Shippable  etc etc.

With Docker technology you can say here is my repo (on github). Take Ubuntu, Tomcat and Java libraries and build me an image. 

I am oversimplify it here. But the overall point is that Docker will make tools like Chef not relavent.

The bet we are going to make is on container / Docker technogy.

Docker is much bigger then replacement to Chef. Docker / Container technology is a replacement for virtual machine. 

When I asked you to use Shippable there was a game plan behind it. As a next step you can ask Shippable to build and image that we can deploy.

It may be confusing initially. But containers are the only way forward to go.

BTW: No body in IBA will teach you this. 😊

Allah Hafiz.

**Comment by sojharo** [comment]
Yes, it looked great when I was learning docker from small tryit tutorial. Let me know if I need to try the ice building on docker. I can create the war file from the ice now. This is the file which can be deployed on the server. Other steps in this is to use chef or docker to deploy this file to TomCat.

**Comment by jekram** [comment]
Docker is the right way to do it. Instead of Chef. Right now we have manual process to deploy. So let's focus on ICE for Azure and Google first then we can come back and look into how we can automate the deployment. When we do it we should do it with Docker.

**Comment by sojharo** [comment]
Ok, so let us keep deployment on hold and I am going to look into ICE azure.

**Comment by jekram** [comment]
Yes. Thanks.

**Comment by jekram** [comment]
I am going to block this issue. We would use Docker to solve deployment.
